src/nbl/video/CVulkanCommandPool.cpp

```diff
@@ -12,6 +12,15 @@ CVulkanCommandPool::~CVulkanCommandPool()
     vk->vk.vkDestroyCommandPool(vulkanDevice->getInternalObject(), m_vkCommandPool, nullptr);
 }
 
+void CVulkanCommandPool::reset()
+{
+	const CVulkanLogicalDevice* vulkanDevice = static_cast<const CVulkanLogicalDevice*>(getOriginDevice());
+	auto* vk = vulkanDevice->getFunctionTable();
+	// TODO work out: always use release resources flag?
+	vk->vk.vkResetCommandPool(vulkanDevice->getInternalObject(), m_vkCommandPool, VK_COMMAND_POOL_RESET_RELEASE_RESOURCES_BIT);
```

Comment:
Ok @deprilula28 I see that you got some enums a bit mixed up here. Let me explani why you don't want the `VK_COMMAND_POOL_RESET_RELEASE_RESOURCES_BIT` here.
1. CommandPool allocates from system memory
2. CommandBuffer allocates from command pool

So reseting the command pool with that flag recycles all of the resources from the command pool back to the system.
But reseting the command buffer with `VK_COMMAND_BUFFER_RESET_RELEASE_RESOURCES_BIT` specifies that most or all memory resources currently owned by the command buffer should be returned to the parent command pool. 

By this definitions reseting a command pool with no additional flags is like reseting every one of the command buffers with `VK_COMMAND_BUFFER_RESET_RELEASE_RESOURCES_BIT`, and that's what we want in the reset command pool. so that each of the  command buffer return their resources back to parent pool

Notice `VK_COMMAND_BUFFER_RESET_RELEASE_RESOURCES_BIT` is different from `VK_COMMAND_POOL_RESET_RELEASE_RESOURCES_BIT`